metadataserver/iam.go

```diff
@@ -347,10 +347,12 @@ func (proxy *iamProxy) doAssumeRole(sessionLifetime time.Duration) {
 }
 
 func (proxy *iamProxy) getRoleAssumptionState(ctx context.Context) *roleAssumptionState {
+	startTime := time.Now()
 	// So this can potentially block for up to the upper bound of the request timeout
 	proxy.startRoleAssumer()
 	proxy.roleAssumptionStateLock.RLock()
 	defer proxy.roleAssumptionStateLock.RUnlock()
+	log.WithFields(logging.Entry(ctx)).WithField("acquire-time", time.Since(startTime).Milliseconds()).Info("got role assumption state")
```

Comment:
This is going to log _a lot_. What do you think about adding tracing here instead?

Reply:
It will log one extra line whenever `/info` and `/security-credentials/{id}` are called.  Is that actually going to be a lot?  I thought about adding tracing, but didn't think 2 extra lines per request were going to be _that_ much.

Reply:
I don't know, I guess I'd rather see if we can add it to the request context and in the middleware that does the request logging, if we can have it passed back in there.